src/eradius_counter.erl

```diff
@@ -76,6 +87,29 @@ aggregate({Servers, {ResetTS, Nass}}) ->
     NSum1 = [Value || {_Key, Value} <- orddict:to_list(NSums)],
     {Servers, {ResetTS, NSum1}}.
 
+%% @doc Update the given histogram metric value
+%% NOTE: We use prometheus_histogram collector here instead of eradius_counter ets table because
+%% it is much easy to use histograms in this way. As we don't need to manage buckets and do
+%% the other histogram things in eradius, but prometheus.erl will do it for us
+observe(Name, {{ClientName, ClientIP, _}, {ServerName, ServerIP, ServerPort}} = MetricsInfo, Value, Help) ->
+    try
+        prometheus_histogram:observe(Name, [ServerIP, ServerPort, ServerName, ClientName, ClientIP], Value)
+    catch _:_ ->
+            Buckets = application:get_env(eradius, histogram_buckets, [10, 30, 50, 75, 100, 1000, 2000]),
+            prometheus_histogram:new([{name, Name}, {labels, [server_ip, server_port, server_name, client_name, client_ip]},
+                                      {buckets, Buckets}, {help, Help}]),
+            observe(Name, MetricsInfo, Value, Help)
+    end.
+observe(Name, #nas_prop{server_ip = ServerIP, server_port = ServerPort, nas_ip = NasIP, nas_id = NasId} = Nas, Value, ServerName, Help) ->
+    try
+        prometheus_histogram:observe(Name, [inet:ntoa(ServerIP), ServerPort, ServerName, inet:ntoa(NasIP), NasId], Value)
+    catch _:_ ->
+            Buckets = application:get_env(eradius, histogram_buckets, [10, 30, 50, 75, 100, 1000, 2000]),
+            prometheus_histogram:new([{name, Name}, {labels, [server_ip, server_port, server_name, nas_ip, nas_id]},
+                                      {buckets, Buckets}, {help, Help}]),
+            observe(Name, Nas, Value, ServerName, Help)
+    end.
+
```

Comment:
eradius is supposed to be library. I would therefore prefer to keep mandatory dependencies as small as possible.

Do think it would be possible to wrap the prometheus calls in a `case code:is_loaded(prometheus)`, move the collector module to a seperate application (maybe in an apps directory in this project) and remove the prometheus dependency from eradius itself?

Reply:
note: prometheus_histogram is used here only to avoid re-implementation of the _bucket, _sum, _total sub metrics. just what the comment states